install/stable/InstallHalyard.sh

```diff
@@ -0,0 +1,366 @@
+                                rather than the default repository, which is
+                                $REPOSITORY_URL.
+
+    --version <version>         Specify the exact verison of Halyard to install.
+
+    --dependencies_only         Do not install any Spinnaker services.
+                                Only install the dependencies. This is intended
+                                for development scenarios only.
+
+    --local-install             For Spinnaker and Java packages, download
+                                packages and install using dpkg instead of
+                                apt. Use this option only if you are having
+                                issues with the bintray repositories.
+                                If you use this option you must manually
+                                install openjdk-8-jdk.
+
+    --home_dir                  Override where user home directories reside
+                                example: /export/home vs /home.
+EOF
+}
+
+function echo_status() {
+  if [ -n "$QUIET" ]; then
+    echo "$@"
+  fi
+}
+
+function process_args() {
+  while [ "$#" -gt "0" ]
+  do
+    local key="$1"
+    shift
+    case $key in
+      --repository)
+        echo "repo"
+        REPOSITORY_URL="$1"
+        shift
+        ;;
+      --version)
+        echo "version"
+        VERSION="$1"
+        shift
+        ;;
+      --dependencies_only)
+        echo "deps"
+        DEPENDENCIES_ONLY=true
+        ;;
+      -y)
+        echo "non-interactive"
+        YES=true
+        ;;
+      --local-install)
+        echo "local"
+        DOWNLOAD=true
+        ;;
+      --quiet|-q)
+        QUIET=true
+        ;;
+      --home_dir)
+        homebase="$1"
+        if [ "$(basename $homebase)" = "spinnaker" ]; then
+          echo "stripping trailing 'spinnaker' from --home_dir=$homebase"
+          homebase=$(dirname $homebase)
+        fi
+        shift
+        ;;
+      --help|-help|-h)
+        print_usage
+        exit 13
+        ;;
+      *)
+        echo "ERROR: Unknown argument '$key'"
+        exit -1
+    esac
+  done
+}
+
+function add_apt_repositories() {
+  # Spinnaker
+  # DL Repo goes here
+  REPOSITORY_HOST=$(echo $REPOSITORY_URL | cut -d/ -f3)
+  if [ "$REPOSITORY_HOST" = "dl.bintray.com" ]; then
+    REPOSITORY_ORG=$(echo $REPOSITORY_URL | cut -d/ -f4)
+    # Personal repositories might not be signed, so conditionally check.
+    gpg=""
+    gpg=$(curl -s -f "https://bintray.com/user/downloadSubjectPublicKey?username=$REPOSITORY_ORG") || true
+    if [ -n "$gpg" ]; then
+      echo "$gpg" | apt-key add -
+    fi
+  fi
+  echo "deb $REPOSITORY_URL ${DISTRIB_CODENAME}-${RELEASE_TRACK} spinnaker" | tee /etc/apt/sources.list.d/halyard.list > /dev/null
+  # Java 8
+  # https://launchpad.net/~openjdk-r/+archive/ubuntu/ppa
+  add-apt-repository -y ppa:openjdk-r/ppa
+  apt-get update ||:
+}
+
+function install_java() {
```

Comment:
Maybe at some point you should have a common bash functions file you can source rather than repeating these functions. That being said, I don't know what's being templated into these install scripts. Just a thought.